axelrod/strategies/mindreader.py

```diff
@@ -46,15 +46,25 @@ class MindReader(Player):
     """A player that looks ahead at what the opponent will do and decides what to do."""
 
     name = 'Mind Reader'
-    stochastic = True # Don't cache me
+    behaviour = {
+        'memory_depth': -10,
+        'stochastic': False,
```

Comment:
What about just changing all of these to be:

```
manipulates_state
inspects_source
manipulates_source
```

?

Any thoughts on this @meatballs ?